openmdao/core/group.py

```diff
@@ -1438,13 +1434,10 @@ def _setup_global_connections(self, conns=None):
 
         # Add implicit connections (only ones owned by this group)
         for prom_name in allprocs_prom2abs_list_out:
-            if prom_name in allprocs_prom2abs_list_in:
+            if prom_name in allprocs_prom2abs_list_in:  # names match ==> a connection
                 abs_out = allprocs_prom2abs_list_out[prom_name][0]
-                out_subsys = abs_out[path_len:].split('.', 1)[0]
                 for abs_in in allprocs_prom2abs_list_in[prom_name]:
-                    in_subsys = abs_in[path_len:].split('.', 1)[0]
-                    if out_subsys != in_subsys:
```

Comment:
Did we have a reason that we originally only added the connection to our dict under this conditional?

Reply:
I'm glad you asked about this, because it made me realize that my first fix wasn't right because it was causing unnecessary transfers.  The pre-existing check *is* necessary to determine which group owns the transfers.   The original code was just failing to add all of the implicit connections to _conn_global_abs_in2out which contains all connections in a group or any group contained in it.